release: 0.1.0 (#500)
axodotdev/oranda: CHANGELOG.md

# Changelog

## 0.1.0 - 2023-06-29

### What is oranda?

> oranda is a hands-off static site generator for people who want a website for their tool but don't want to learn a static site generator. As long as you have a `README.md` in your directory, you can benefit from oranda. oranda will also try to automatically work with:

Reviewer: ```suggestion
oranda is a hands-off static site generator for people who want a website for their tool but don't want to get knee-deep into web development. As long as you have a `README.md` in your directory, you can benefit from oranda. oranda will also try to automatically work with:
```

- Release artifacts (currently only for GitHub releases)
    - `cargo-dist`-generated ones, and
    - arbitrary release artifacts
- `mdbook`docs
- GitHub-supported project/maintainer funding sources

> oranda is designed to _just work_ in a lot of cases, and in cases where it doesn't, to provide fine-grained configuration so you can make it work for you.

Reviewer: ```suggestion
oranda is designed to _just work_ in a lot of cases, and in cases where it doesn't, it should provide fine-grained configuration so you can make it work for your use case.
```

### Features

#### Components

##### Github release artifacts inference

> We can now not only figure out whether you're using `cargo-dist` _automatically_, but we also try and support arbitrary tarballs, as long as they're attached to a release and they're following the target-triple format. Oranda will now, for example, pick up a release artifacts called `myapp-aarch64-apple-darwin.tar.xz`, even if the project isn't using `cargo-dist`.

Reviewer: ```suggestion
We can now not only figure out whether you're using `cargo-dist` _automatically_, but we also try and support arbitrary tarballs, as long as they're attached to a release and they're following the target-triple format. Oranda will now, for example, pick up a release artifact called `myapp-aarch64-apple-darwin.tar.xz`, even if the project isn't using `cargo-dist` to publish releases.
```

##### Smarter install widget

> The installer widget on our main page has been upgraded! It now not only shows a select box where you can switch between different architectures (though we still attempt to figure out what platform you're running on), but it now additionally displays your package managers:

Reviewer: ```suggestion
The installer widget on our main page has been upgraded! It now not only shows a select box where you can switch between different architectures (though we still attempt to figure out what platform you're running on), but it now additionally displays your package managers, all in a sleek tab-style view:
```

> ![](https://hackmd.io/_uploads/B1iKMxju3.png)

Reviewer: ```suggestion
![oranda install widget preview](https://github-production-user-asset-6210df.s3.amazonaws.com/6445316/250566861-a635c28b-d4d3-4c90-a685-8e9a85673651.png)
```

> You can customize which package managers you want to be displayed on in this widget vs. which ones should only be displayed on the separate install page, as well.

Reviewer: ```suggestion
You can customize which package managers you want to be displayed on in this widget vs. which ones should only be displayed on the separate install page, as well. [Read more in the docs](https://opensource.axo.dev/oranda/book/configuration/artifacts.html#adding-package-manager-installation-instructions)
```

##### Funding page

oranda now has the ability to autodetect whether you're using GitHub's funding embed functionality (meaning you have a `.github/FUNDING.yml`), in which case it'll automatically generate a page showing your available funding options.

> Additionally, you can enhance this page by selecting a particular funding channel to be showcased as well as providing custom content from a `funding.md` Markdown file. You can, of course, fully disable this feature, or adjust the paths if you want.

Reviewer: ```suggestion
Additionally, you can enhance this page by selecting a particular funding channel to be prioritized as your "main" funding method. You can also provide custom content from a `funding.md` Markdown file, to provide additional context to your funding page. [Read more in the docs](https://opensource.axo.dev/oranda/book/configuration/funding.html)
```

Author: for this one, i was wondering if instead of highlighting that you can turn it off- maybe we highlight that you can prioritize a particular method, etc

Reviewer: i think that's what the beginning of this paragraph is supposed to say lol let me adjust my suggestion

Reviewer: does this look any better?
